spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/source/SparkBatchQueryScan.java

```diff
@@ -256,4 +275,56 @@ public String toString() {
         runtimeFilterExpressions,
         caseSensitive());
   }
+
+  @Override
+  public CustomTaskMetric[] reportDriverMetrics() {
+    List<CustomTaskMetric> customTaskMetrics = Lists.newArrayList();
+    Optional<ScanReport> scanReportOptional = sparkReadMetricReporter.getScanReport();
+
+    scanReportOptional.ifPresent(
+        scanReport -> {
+          Optional.ofNullable(scanReport.scanMetrics().totalFileSizeInBytes())
+              .ifPresent(
+                  counterResult ->
+                      customTaskMetrics.add(new TaskTotalFileSize(counterResult.value())));
+          Optional.ofNullable(scanReport.scanMetrics().totalPlanningDuration())
+              .ifPresent(
+                  timerResult ->
+                      customTaskMetrics.add(new TaskTotalPlanningDuration(timerResult.count())));
+
+          Optional.ofNullable(scanReport.scanMetrics().skippedDataFiles())
+              .ifPresent(
+                  skippedDataFilesResult ->
+                      customTaskMetrics.add(
+                          new TaskSkippedDataFiles(skippedDataFilesResult.value())));
+          Optional.ofNullable(scanReport.scanMetrics().resultDataFiles())
+              .ifPresent(
+                  resultDataFilesResult -> new TaskResultDataFiles(resultDataFilesResult.value()));
+
+          Optional.ofNullable(scanReport.scanMetrics().skippedDataManifests())
+              .ifPresent(
+                  skippedDataManifestsResult ->
+                      customTaskMetrics.add(
+                          new TaskSkippedDataManifests(skippedDataManifestsResult.value())));
+          Optional.ofNullable(scanReport.scanMetrics().scannedDataManifests())
+              .ifPresent(
+                  scannedDataManifestResult ->
+                      customTaskMetrics.add(
+                          new TaskScannedDataManifests(scannedDataManifestResult.value())));
+        });
+
+    return customTaskMetrics.toArray(new CustomTaskMetric[0]);
+  }
+
+  @Override
+  public CustomMetric[] supportedCustomMetrics() {
```

Comment:
This overrides `supportedCustomMetrics()` in `SparkScan` and breaks it. Can we move this logic there and combine with existing metrics?